### ISSUE TITLE
Add MathML Block

### ISSUE THREAD

**Issue by EldarAgalarov** [WordPress/gutenberg#47198]
Hi. It would be nice to have support for MathML blocks with basic functionality such as formula size, centering, etc.
Now to insert MathML code the Custom HTML block should be used.

**Comment by kathrynwp** [comment]
Hi @EldarAgalarov have you given this plugin a try to add a MathML block?

https://wordpress.org/plugins/mathml-block/

I'm not sure enough people would use this block for it to be considered for inclusion in Core; the plugin has only 900 active installs.

**Comment by EldarAgalarov** [comment]
@kathrynwp Yes, I know about such plugins, but I prefer to be plugin-independent as much as possible, so i'm using just Custom HTML block at this moment :) MathML is part of HTML 5 standard so it sounds logical that Math ML Block should exists in the Gutenberg, like Heading, Paragraph, Table, Image blocks.

**Comment by dmsnell** [comment]
Having poked around on a [math typesetting block](https://fluffyandflakey.blog/2019/09/09/typesetting-math-in-gutenberg/) myself, I'll share where I kind of left it off, and unfortunately why I think "MathML" isn't well-suited for core (hopefully "yet").

In short, MathML support in browsers is still somewhat abysmal. @adamsilverstein's block loads after page render to handle this by rendering the MathML via JavaScript; mine did the same thing but stored the generated rendered XML as a data URI to avoid a flash on initial page load (and to avoid pushing the cost of rendering to the client).

I would love a core formula block, but I think it'll be complicated by needing to serve a lot of different needs, which is good! Some people want to see a formula-builder UI so all they need to do is click buttons to generate a formula; others want LaTex syntax to quickly enter formulas as text. All that is fine I think, but needs people to do the work, plus it involves pulling in third-party libraries to parse the input formulas or present the UI (unless we want to write and maintain our own formula entry and rendering code).

So with the plugin space at least for now it's possible to choose your poison, so to speak, which might be much easier in the meantime than finding something that works for everyone. As I have a personal interest in this I'd be happy to review any code submissions or help guide along the project, but at the moment I'm not likely able to dedicate my time to creating such a block.

**Comment by adamsilverstein** [comment]
One great feature of the block inserter is the plugin search. If you type in /mathml the inserter will offer up the plugin and let you use it immediately. So using mathml currently is very low friction, although I support the idea of using native mathml support - and browser support is improving.

**Comment by EldarAgalarov** [comment]
> In short, MathML support in browsers is still somewhat abysmal

Firefox supporting MathML for a long time
Chromium added MathML back since 109

**Comment by dmsnell** [comment]
Agreed that Firefox's support looks pretty good, but a couple examples from Mozilla's test page shows some significant problems for Safari and for the Chrome version I have installed, which is 107.

In the following screenshots we have a LaTeX render on the left, an image of Firefox render in the middle, and the actual browser render on the right.

Essentially nothing in Chrome
<img width="1530" alt="Screenshot 2023-01-18 at 2 12 59 PM" src="https://user-images.githubusercontent.com/5431237/213338560-a4538fc1-6585-4c74-814f-ef34d1393cae.png">

Less severe problems here but still not a good experience, and these aren't exposed on particularly complicated formulas either.
<img width="1192" alt="Screenshot 2023-01-18 at 2 13 33 PM" src="https://user-images.githubusercontent.com/5431237/213353066-266322ca-29ca-48eb-aec4-6a1a5e4c6f1d.png">
<img width="1182" alt="Screenshot 2023-01-18 at 2 13 42 PM" src="https://user-images.githubusercontent.com/5431237/213353074-e2bd84b3-4201-40e3-88f2-945c24953922.png">

So even though Chrome has finally added support back in for MathML in their latest release (which was released just a week ago), that still leaves a pretty large group of readers behind and will for some time.

**Comment by EldarAgalarov** [comment]
> Chrome version I have installed, which is 107

You should install version 109 and then you will get following:
![image](https://user-images.githubusercontent.com/11322690/213357524-4b6299d4-b03b-435d-9ffe-056e3b48f234.png)
![image](https://user-images.githubusercontent.com/11322690/213360040-db0fd149-b80e-4b96-a715-2ed6b3ed7c57.png)

Still worse than Firefox rendering, but much better than "rendering" of previous versions of Chromium :)

**Comment by bgoewert** [comment]
Somewhat related Trac ticket: https://core.trac.wordpress.org/ticket/13340

**Comment by ellatrix** [comment]
See also #35005. I think this would be interesting inline as well.

**Comment by mtias** [comment]
I think we should revisit this one now. It'd be great to have awesome math support in core.

**Comment by dmsnell** [comment]
Since 2023 support has gotten significantly better, probably to the point where we can produce raw MathML output. Core will still fight it, but we also now have the HTML API which understands the complexity of MathML embedded within HTML, which itself is much more complicated than a standalone MathML XML file.

Further, if we store the MathML as a data URI, encoded as base64, we should be able to bypass Core mangling it.

<table><tr><th>Safari<th>Firefox<th>Chrome><tr><td><img width="1204" height="868" alt="Image" src="https://github.com/user-attachments/assets/a2e2424d-8bb7-4563-8d75-ba671423ab06" />
<td><img width="1194" height="925" alt="Image" src="https://github.com/user-attachments/assets/5a30e74b-54c0-4d17-9dbc-0d4740498d86" />
<td><img width="1178" height="926" alt="Image" src="https://github.com/user-attachments/assets/115967a2-5e37-4f3c-b0b2-ceff206520f0" />

**Comment by ellatrix** [comment]
Nice! IMO we should definitely do MathML then, putting this in WP will also push browser to fix issues and not withdraw support.
